bpf/lib/nat.h

```diff
@@ -1720,9 +1717,10 @@ snat_v6_needs_masquerade(struct __ctx_buff *ctx __maybe_unused,
 		 * is on 4 (after pfx.lpm.prefixlen), and we can't use memcpy()
 		 * on the whole field or the verifier complains.
 		 */
-		memcpy(pfx.lpm.data, &ip6->daddr, 4);
-		memcpy(pfx.lpm.data + 4, (__u8 *)&ip6->daddr + 4, 8);
-		memcpy(pfx.lpm.data + 12, (__u8 *)&ip6->daddr + 12, 4);
+		memcpy(pfx.lpm.data, &tuple->daddr.p1, 4);
+		memcpy(pfx.lpm.data + 4, &tuple->daddr.p2, 4);
+		memcpy(pfx.lpm.data + 8, &tuple->daddr.p3, 4);
+		memcpy(pfx.lpm.data + 12, &tuple->daddr.p4, 4);
```

Comment:
The comment above doesn't make sense after this change. Can we leave that optimization here? If not, then we need to remove the comment.

Reply:
Ok, this works. (I also checkedm just in case, that this will translate to 4 `u32` writes and tnothing more.)